lib/pacemaker/pcmk_sched_resource.c

```diff
@@ -384,13 +384,11 @@ add_assigned_resource(pe_node_t *node, pe_resource_t *rsc)
  *       from calling pcmk__unassign_resource(), in that it will also update any
  *       actions created for the resource.
  */
-bool
-pcmk__finalize_assignment(pe_resource_t *rsc, pe_node_t *chosen, bool force)
+static bool
+assign_resource(pe_resource_t *rsc, pe_node_t *chosen, bool force)
```

Comment:
I took "finalize" out of the name proactively, because we'll need it to be fully reversible in order to fix [T489](https://projects.clusterlabs.org/T489).

There's currently no way to undo an assignment if `chosen == NULL` or if we fail to find an available node. `pcmk__finalize_assignment()` sets `rsc->next_role` to Stopped in that case and updates its existing actions. `pcmk__unassign_resource()` has no way to undo that.

I have another commit queued up locally that adds a flag for this. I'm saving it until the T489 fix is ready, and that in turn depends on T672.